python/tank/platform/events/event_engine.py

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+class EngineEvent(object):
+    """
+    The base class of all concrete engine event objects. It provides a very
+    basic interface that is expanded on in deriving classes.
+    """
+    _EVENT_TYPE = None
```

Comment:
quick thought - do we need this now that we have the subclass approach? Feels like we can drop it - isinstance feels clearer to me (and you even recommend it in the docs!).